.travis.yml

```diff
@@ -0,0 +1,232 @@
+  - name: deployment
+    if: tag =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc|alpha|dev)([0-9]+$))?|(-git[0-9a-z]{6,8})?$
+
+.docs: &docs_cond
+  if: ( ( branch = master ) OR ( tag =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc|alpha|dev)([0-9]+$))?|(-git[0-9a-z]{6,8})?$ ) )
+.draftrelease: &draftrelease_cond
+  condition: $TRAVIS_TAG =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc|dev|alpha)([0-9]+))?(-git[0-9a-z]{6,8})$
+.prerelease:   &prerelease_cond
+  condition: $TRAVIS_TAG =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc)([0-9]+))$
+.devrelease:   &devrelease_cond
+  condition: $TRAVIS_TAG =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(dev|alpha)([0-9]+))$
+.release:      &release_cond
+  condition: $TRAVIS_TAG =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)$
+# .deploy:       &deploy_cond
+#   condition: $TRAVIS_TAG =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc|alpha|dev)([0-9]+$))?|(-git[0-9a-z]{6,8})?$
+.deploy:       &deploy_cond
+  if: tag =~ ^v?(.)?([0-9]+).([0-9]+).([0-9]+)(-(pre|rc|alpha|dev)([0-9]+$))?|(-git[0-9a-z]{6,8})?$
+
+env:
+  global:
+    ## Encrypted variables:
+    ##  * CHANGELOG_GITHUB_TOKEN: for running github_changelog_generator without API rate limits
+    # - secure: i4K316XPS0ANZeU6bF0JIsxZOaD/7gD5IwXR+kQwbENLpf6DsrLTE6ZJKgS/r9jClOAvbAQj9R7O61uRCHPYzQemKkXhCLSlqMlk0/RdvLVlcFCR6C7SDPF5DKnA8QQ3r7c2jdv27/TFfk60VWOYbQln94Mh8n3kH+6NGOwFwjxLBce88v8lsZXb3NDiHzwxXMeTbqqrAAbhCrjr7hrIvpm7eJwXVhOCz2HR7SpDOIFhwaDgGNgotqleF2szXwVnOUE4RxOgR7zBKLLFwjOtcdS0nRvOctf7l6kiq0t16V+BRIr0kad9hMUscNEKHDmQmRWF9zI83k2j/Ux4kgq1NUy6Jx+fbs/Y0WIwrd+LCyGIuaw/Y+Y3LVwT1EmUzYoga7oq2VB7j07PrOvGa4BwFDN1zm+7n2EZtmDw5aZwTtMeCkm4L+dL8HoiVVWAaFMZ8l/EYDDqWea5dcg9uazdI/wUFLavuc1lRzGTGHK+fxEYmX5DOfK35nyJQqDjB6dCKFjsty9NR1nF1wnsk49IuvviN6IfpCiIaGTxgAUrV8mT32gxPB9DdDyiPsQXHS6Z8ha2aq9Eeq7Zm/TXA5WkbUcw3RJsmdX6oDj89FzElX5EK0KjwRJgqgFfDP7yIhQHsRSk0QnWhIO6YI78A30DaTwR7vKNex09eT3igm+4ppU=
+    - secure: KJbkEj0oKPEdHIAemqb9724uG0G3DDahzpl0ExEkQy/u5Gfkt5QgfFx8Ft5gzZtBkecLoC9weuUMzlFE4UEOi41alxZsJ8kKJCpPV8d0zzzQ7aDR8tpVE7CbCf0r7sScsq5sLXiXHgY5QMDRKjx40a+48cd3WrnGtoGDw7AKWiFS5hg/9fhZzwxnSlYALyMNYNnnaITsiBRlllRKIiEzBHqQOtCxHXKMLGgetwNnw58oRD64F6+l/aqIWAnrjZAdKtH4EUIIPJOEr2zWvlFQzIFzPLAo7O4ygVhOEituF/149s4/uPRWv06BcEQ0NIqP3qppT//voI832xaiQPrY2oiBD8lDoDiVfFKcM9lWzhtaxoJKRQ19DJGO9E0yXstwwjE/UOMJoYU4x6wdMbx7Tu6wlFgvxjBb56Pufvanq5QwsfioOU/aIvR75bPn6CfsYED8UXIHuOmRtKo41ApXru2OfqjAP2d67rBjW60aTwLv8J36zH1497DunbFkckqSOIHhi7MqUNn4+ghZGWVYMKXBmqLecW8hmdbHq2Jw95ZHDaHX8Ve4Z5LQ+0VF2Ydas9ULBQjH5viMDpRPURfpxlDGfgBe1E4rdZXrUpPpyJejumD8qJCHLwuHRjLKhBFUOuuyEGnQ1NVZN80LK+Xzw4Faujvkb9QY2npeGNVSTIY=
+
+  matrix:  ## this is *only* for the built-in 'test' stage, doesn't currently transfer to other stages, beta feature
+    - OS_TYPE=centos PY_VER=python OS_VERSION=6 DOCKER_IMAGE=gitlab-registry.cern.ch/sturdy/gemdaq_ci_worker/extrapy/withroot:slc6 ROOT_VER=v5.34.36-gcc4.4
+    # - OS_TYPE=centos PY_VER=python2.6 OS_VERSION=6 DOCKER_IMAGE=gitlab-registry.cern.ch/sturdy/gemdaq_ci_worker/extrapy/withroot:slc6
```

Comment:
Why do we have jsturdy personal docker image here? Can't we have a common one? Also do we need to keep the commented lines below?

Reply:
* needs a change post merge
* they can also be cleaned up, but if we want to test against different arch/python/root versions, this was a template of one possible way

Reply:
Seems like testing is right now on `slc6` with root `v5.34` however the development machines are all on `cc7` (`python 2.7`) the production machine on `slc6` (`python 2.6`).  Do we plan to change this in the near term future or should development be done with this version of root and python in mind?

Reply:
Or you just plan on uncommenting a given line...? Sorry not sure how all the bells and whistles fit together yet.  Probably this comes at a next step when instructions on how to rebase are provided.